File: languages/js/src/entity/identity_v2/datetime_attribute.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities_v2";
import { Attribute } from "./attribute";

export class DateTimeAttribute extends Attribute<Date> {
  public toProto(): identityEntitiesProto.DateTimeAttributeV2 {
    return identityEntitiesProto.DateTimeAttributeV2.fromPartial({
      id: this.id,
      value: this.value.toISOString()
    });
  }

  static fromProto(
    r: identityEntitiesProto.DateTimeAttributeV2
  ): DateTimeAttribute {
    return new DateTimeAttribute(r.id, new Date(r.value));
  }
}

File: languages/python/bloock/entity/identity_v2/datetime_attribute.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_v2_pb2 as proto
from bloock.entity.identity_v2.attribute import Attribute


class DatetimeAttribute(Attribute):
    """
    Represents an attribute with a datetime value.
    """
    @staticmethod
    def from_proto(a: proto.DateTimeAttributeV2) -> DatetimeAttribute:
        return DatetimeAttribute(
            id=a.id,
            value=a.value,
        )

    def to_proto(self) -> proto.DateTimeAttributeV2:
        return proto.DateTimeAttributeV2(
            id=self.id,
            value=self.value,
        )